registry.go

```diff
@@ -11,12 +11,30 @@ type RegistryClient struct {
 	*Client
 }
 
+// CreateRegistry
+// Create a registry
+func (s *RegistryClient) CreateRegistry(r Registry) error {
+	resp, _, errs := s.NewRequest(gorequest.POST, "").
+		Send(r).
+		End()
+	return CheckResponse(errs, resp, 201)
+}
+
+// ListRegistries
+// Get a list of registries
+func (s *RegistryClient) ListRegistries(name string) ([]Registry, error) {
+	var r []Registry
+	resp, _, errs := s.NewRequest(gorequest.GET, "").
+		Query(map[string]string{"name": name}).
```

Comment:
What does the API respond with, if we just pass `&name=` as query? Maybe we should consider `name == ""` as a special case in which no query should be added at all (to list _all_ registries)?

Reply:
Good point, the API seems to not require a query string on this endpoint.
When leaving it out entirely / querying any unsupported key, the response stays the same (an array of all registries):

```
/api/registries
/api/registries?foo
/api/registries?name=
```

Other endpoints show this behaviour as well, e.g. `/api/replication/policies` acts the same.

IMHO, i think we don't need to catch this here, but we should let the user know about this quirk through documentation. Your thoughts?

Reply:
Probably doesn't make that much of a difference. 🤷 I think the general behaviour (leave `name` empty, get all registries) is valid; we _might_ go the extra step of removing the query parameter entirely if the user supplies an empty `name` query, in case the server ever decides to change its default behaviour (server-side, they're probably doing the same as us: decoding into a Go `string` and making no difference between "set, but empty" and "not set at all").

TL;DR: It's fine by me to leave it as it is.